src/tools/interop/idt/capture/controller.py

```diff
@@ -0,0 +1,179 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import asyncio
+import copy
+import os
+import sys
+import traceback
+import typing
+
+import capture
+from capture.base import EcosystemCapture, PlatformLogStreamer, UnsupportedCapturePlatformException
+from utils.artifact import create_standard_log_name, log, safe_mkdir
+from utils.log import border_print
+
+from . import config
+
+_PLATFORM_MAP: typing.Dict[str, PlatformLogStreamer] = {}
+_ECOSYSTEM_MAP: typing.Dict[str, PlatformLogStreamer] = {}
+_ERROR_REPORT: typing.Dict[str, list[(str, str, str)]] = {}
```

Comment:
tuple of 3 strings is not as friendly to use. How about a `@dataclass` with 3 memebers of error_type, error_message and a traceback (by the way, I think the traceback is not a string)

Reply:
Thanks i implemented the dataclass cleaned up the error reporting overall. 

I also removed the error type abstraction that wasn't really serving any purpose. 

I believe format_exec() returns a string based on https://docs.python.org/3/library/traceback.html#traceback.format_exc 

And in my IDE

```python3
def format_exc(limit: int | None = ...,
               chain: bool = ...) -> str
```

so I left that

I tested it with some temp forced failures in playservices and playservices user